docs/en/query_language/agg_functions/reference.md

```diff
@@ -361,45 +404,119 @@ Calculates the average.
 
-## uniqHLL12(x)
+- Provides the determinate result (it doesn't depend on the order of query processing).
 
-Uses the [HyperLogLog](https://en.wikipedia.org/wiki/HyperLogLog) algorithm to approximate the number of different values of the argument.
-212 5-bit cells are used. The size of the state is slightly more than 2.5 KB. The result is not very accurate (up to ~10% error) for small data sets (<10K elements). However, the result is fairly accurate for high-cardinality data sets (10K-100M), with a maximum error of ~1.6%. Starting from 100M, the estimation error increases, and the function will return very inaccurate results for data sets with extremely high cardinality (1B+ elements).
+We recommend to use this function in almost all scenarios.
 
-The result is determinate (it doesn't depend on the order of query processing).
 
-We don't recommend using this function. In most cases, use the `uniq` or `uniqCombined` function.
+## uniqCombined {#agg_function-uniqcombined}
+
+Calculates the approximate number of different values of the argument.
+
+```
+uniqCombined(HLL_precision)(x[, ...])
+```
+
+The `uniqCombined` function is a good choice for calculating the number of different values, but keep in mind that the estimation error for large sets (200 million elements and more) will become larger than theoretical value due to poor choice of hash function.
+
+**Parameters**
+
+Function takes the variable number of parameters. Parameters can be of types: `Tuple`, `Date`, `DateTime`, `String`, numeric types.
+
+`HLL_precision` is the base-2 logarithm of the number of cells in [HyperLogLog](https://en.wikipedia.org/wiki/HyperLogLog). Optional, you can use the function as `uniqCombined(x[, ...])`. The default value for `HLL_precision` is 17, that is effectively 96 KiB of space (2^17 cells of 6 bits each).
 
-## uniqExact(x)
+**Returned value**
 
-Calculates the number of different values of the argument, exactly.
-There is no reason to fear approximations. It's better to use the `uniq` function.
-Use the `uniqExact` function if you definitely need an exact result.
+- The number of the [UInt64](../../data_types/int_uint.md) type.
+
+**Implementation details**
+
+Function:
+
+- Calculates a hash for all parameters in the aggregate, then uses it in calculations.
+- Uses combination of three algorithms: array, hash table and HyperLogLog with an error correction table.
+
+    For small number of distinct elements, the array is used. When the set size becomes larger the hash table is used, while it is smaller than HyperLogLog data structure. For larger number of elements, the HyperLogLog is used, and it will occupy fixed amount of memory.
+
+- Provides the determinate result (it doesn't depend on the order of query processing).
+
+In comparison with the [uniq](#agg_function-uniq) function the `uniqCombined`:
+
+- Consumes in several times smaller memory
+- Calculates with the accuracy in several times higher.
```

Comment:
* with several times higher accuracy
* in English "several" means prettify specific range of "3-5 times", is it intended?

Reply:
Yes :-).